From b5a2ac2715f9b3cce04f496d106374492a4dd947 Mon Sep 17 00:00:00 2001
From: thewisenerd <thewisenerd@protonmail.com>
Date: Sat, 24 Dec 2016 07:04:28 +0530
Subject: [PATCH 3/5] firejail: private-* : check, then assign.

---
 src/firejail/main.c | 16 ++++++++--------
 1 file changed, 8 insertions(+), 8 deletions(-)

diff --git a/src/firejail/main.c b/src/firejail/main.c
index f9742cc..c7470c3 100644
--- a/src/firejail/main.c
+++ b/src/firejail/main.c
@@ -1521,38 +1521,38 @@ int main(int argc, char **argv) {
 			}
 			
 			// extract private etc list
-			cfg.etc_private_keep = argv[i] + 14;
-			if (*cfg.etc_private_keep == '\0') {
+			if (*(argv[i] + 14) == '\0') {
 				fprintf(stderr, "Error: invalid private-etc option\n");
 				exit(1);
 			}
+			cfg.etc_private_keep = argv[i] + 14;
 			arg_private_etc = 1;
 		}
 		else if (strncmp(argv[i], "--private-opt=", 14) == 0) {
 			// extract private opt list
-			cfg.opt_private_keep = argv[i] + 14;
-			if (*cfg.opt_private_keep == '\0') {
+			if (*(argv[i] + 14) == '\0') {
 				fprintf(stderr, "Error: invalid private-opt option\n");
 				exit(1);
 			}
+			cfg.opt_private_keep = argv[i] + 14;
 			arg_private_opt = 1;
 		}
 		else if (strncmp(argv[i], "--private-srv=", 14) == 0) {
 			// extract private srv list
-			cfg.srv_private_keep = argv[i] + 14;
-			if (*cfg.srv_private_keep == '\0') {
+			if (*(argv[i] + 14) == '\0') {
 				fprintf(stderr, "Error: invalid private-etc option\n");
 				exit(1);
 			}
+			cfg.srv_private_keep = argv[i] + 14;
 			arg_private_srv = 1;
 		}
 		else if (strncmp(argv[i], "--private-bin=", 14) == 0) {
 			// extract private bin list
-			cfg.bin_private_keep = argv[i] + 14;
-			if (*cfg.bin_private_keep == '\0') {
+			if (*(argv[i] + 14) == '\0') {
 				fprintf(stderr, "Error: invalid private-bin option\n");
 				exit(1);
 			}
+			cfg.bin_private_keep = argv[i] + 14;
 			arg_private_bin = 1;
 		}
 		else if (strcmp(argv[i], "--private-tmp") == 0) {
-- 
2.10.2

